src/components/InvoiceDatasheet/components/SelectEditor.jsx

```diff
@@ -12,15 +13,23 @@ const customStyles = {
   })
 };
 
-const SelectEditor = ({ value, options, onCommit }) => {
-  const handleChange = ({ value }) => {
+const SelectEditor = ({ value, options, onChange, onCommit }) => {
+  const [newValue, setNewValue] = useAsyncState({ value });
```

Comment:
Maybe we could use `getValueObj` to get initial value object ?
```suggestion
  const [newValue, setNewValue] = useAsyncState(getValueObj(value));
```